### PR TITLE
Fix handling of !!map and !!set for empty values

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js [hunk 1]
```diff
@@ -318,54 +378,75 @@ describe('#parseYAML', () => {
...
-    expect(warning).to.have.sourceMapEndLine(3);
-    expect(warning).to.have.sourceMapEndColumn(8);
+  describe('set', () => {
+    it('can parse a set into an array element', () => {
+      const element = parseYAML('!!set\n  - one\n  - two', context);
+
+      expect(element).to.be.instanceof(namespace.elements.ParseResult);
+      expect(element.length).to.equal(2);
+
+      const arry = element.first;
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js]: lowercased `array` isn't a keyword in JavaScript

Comment (honzajavorek) [reply packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js]: or is it a typo?

File: packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js [hunk 2]
```diff
@@ -128,44 +128,104 @@ describe('#parseYAML', () => {
...
+      expect(member.value.toValue()).to.equal('value');
+      expect(member.value).to.have.sourceMapStart(5);
+      expect(member.value).to.have.sourceMapOffset(5);
+      expect(member.value).to.have.sourceMapStartLine(1);
+      expect(member.value).to.have.sourceMapStartColumn(6);
+      expect(member.value).to.have.sourceMapEndLine(1);
+      expect(member.value).to.have.sourceMapEndColumn(11);
+    });
+
+    it('can parse an map into an object element', () => {
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js]:
```suggestion
    it('can parse a map into an object element', () => {
```

File: packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js (hunk 1 above)
Comment (kylef) [reply packages/fury-adapter-oas3-parser/test/unit/parser/parseYAML-test.js]: This was the code before that I just indented, I'll change the occurance of the word

**Pull request merged**